### DISCUSSION TITLE
remark-validate-links and paths

### DISCUSSION THREAD

**Discussion by undavide** [remarkjs/remark#1385] [Q&A]
Hi there,
I am running the following npm script: `"test:links": remark src/pages --quiet --frail"` in a Gatsby based repo, with also:

```
"remarkConfig": {
  "plugins": [
    "remark-validate-links"
  ]
}
```

With `remark-validate-links` v12.x, links that pointed to files within `src/pages` could be written as absolute paths. For instance, given this folder structure:
```
.
└── src
    └── pages
        ├── guides
        │   └── index.md
        └── references
            └── index.md
```

A link in `src/pages/guides/index.md` could be formed either as: 
- `[References](/references/index.md)`
- `[References](../references/index.md)`

I.e., `src/pages` was implied. Starting from v13.x, absolute paths throw a warning. This is just a simple example, but with more complex folder structures, you're drowned in `../../../..`, and as soon as I updated the package I got ~200 new warnings. Please note, Gatsby can manage both urls without a problem.

I tried (unsuccessfully) to use `urlConfig` to get back to the old behavior, I'm wondering whether it's even possible.
Thanks!

Davide

**Comment by wooorm** [comment]
Hi!

This seems to be https://github.com/remarkjs/remark-validate-links/issues/78.

> Please note, Gatsby can manage both urls without a problem.

That is indeed the problem: Gatsby has a particular URL transform for how it does websites.
This project has no clue how that transform works.
This project is focussed on how GitHub/GitLab/Bitbucket and similar work with *Git* repos.
It is not for Gatsby.

More on this can be found in the readme: https://github.com/remarkjs/remark-validate-links#when-should-i-use-this.

It sounds like you are looking more for https://github.com/remarkjs/remark-lint-no-dead-urls.
Importantly though, to have it understand URLs relative to the page, you need to somehow implement Gatsby’s transform, and expose where markdown files “end up” as `origin` and `pathname` on `file.data.meta`.

**Comment by undavide** [comment]
Hi @wooorm,
thanks for the detailed answer. I see your point, and I'll check out `remark-lint-no-dead-urls`.
~I'm still slightly puzzled as to why v12 was Gatsby-friendly, and v13 isn't anymore. Maybe the line between bug and feature is very fine indeed~ (just read the changelog)
